src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

```diff
@@ -105,11 +109,13 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
 
         public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId, string message, Exception exception)
         {
+            if (_sb == null)
```

Comment:
You should cache this in a local, and document that this is safe while we're non-reentrant.

You should also probably go to the trouble of doing acquire/release in case someone subclasses this and does something crazy.

```
var sb = _sb;
_sb = null;

if (sb == null)
{
  _sb == new StringBuilder();
}

.... do logging stuff ...

_sb = sb;
```